src/bin/pg_autoctl/pgsetup.c

```diff
@@ -153,6 +158,13 @@ pg_setup_init(PostgresSetup *pgSetup,
 		}
 		else
 		{
+			/* get the real path of PGDATA now */
+			if (!normalize_filename(pgSetup->pgdata, pgSetup->pgdata, MAXPGPATH))
+			{
+				/* errors have already been logged */
+				return false;
+			}
```

Comment:
Why was the move of this block needed?

Reply:
Not a big deal, could be done otherwise. I had it that way in the other branch where this comes from: we normalise the path after making sure that `pg_controldata` is happy with the PGDATA in this version...